nexus/src/app/utilization.rs

```diff
@@ -42,6 +40,9 @@ impl super::Nexus {
         let (.., authz_pool) =
             pool_lookup.lookup_for(authz::Action::Read).await?;
 
+        // TODO-correctness: We probably want to run these in the same
+        // transaction, otherwise the two counts may change relative to one
+        // another in between each.
```

Comment:
I see that this could happen. But is it detectable by users? I guess we could report some specific pair of (capacity, remaning) that was never true, but could someone tell? And/or would there be any consequences?

I guess that if we get the allocated count and it's 10, and then someone turns off an instance and shrinks the pool, and then we get the capacity we could end up with remaining < 0? Sounds unlikely.

Is there a more robust way to handle this?

Reply:
Yeah, the right way would be to do both queries in a single transaction. It's not too tricky.